grants_tagger_light/retagging/retagging.py

```diff
@@ -0,0 +1,319 @@
+import json
+import logging
+import random
+import time
+
+import typer
+from loguru import logger
+
+from datasets import Dataset, load_dataset, concatenate_datasets
+
+import os
+
+from sklearn import preprocessing
+
+from grants_tagger_light.models.xlinear import MeshXLinear
+from grants_tagger_light.utils.years_tags_parser import parse_years, parse_tags
+import scipy
+import pickle as pkl
+
+import numpy as np
+import tqdm
+
+retag_app = typer.Typer()
+
+
+def _load_data(dset: Dataset, tag, limit=100, split=0.8):
+    """Load data from the IMDB dataset."""
+    min_limit = min(len(dset), limit)
+    dset = dset.select([x for x in range(limit)])
+    # Not in parallel since the data is very small and it's worse to divide and conquer
+    dset.map(
+        lambda x: {"featured_tag": tag},
+        desc=f"Adding featured tag ({tag})",
+    )
+    train_size = int(split * min_limit)
+    train_dset = dset.select([x for x in range(train_size)])
+    test_dset = dset.select([x for x in range(train_size, min_limit)])
+    return train_dset, test_dset
+
+
+def _annotate(curation_file, dset, tag, limit, is_positive):
+    field = "positive" if is_positive else "negative"
+    human_supervision = {tag: {"positive": [], "negative": []}}
+    if os.path.isfile(curation_file):
+        prompt = (
+            f"File `{curation_file}` found. Do you want to reuse previous work? [y|n]: "
+        )
+        answer = input(prompt)
+        while answer not in ["y", "n"]:
+            answer = input(prompt)
```

Comment:
it might be better to throw a message that the answer is not one of the valid ones

Reply:
Done